ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-cpou-upgrade-from-4.10.yaml

```diff
@@ -48,6 +48,12 @@ releases:
       architecture: amd64
       channel: candidate
       version: "4.10"
+  source:
+    candidate:
+      architecture: amd64
+      product: ocp
+      stream: nightly
+      version: "4.11"
```

Comment:
```suggestion
      version: "4.10"
```

Reply:
@jiajliu 
no.
here is the definition
```yaml
releases:
  intermediate:
    candidate:
      architecture: amd64
      product: ocp
      stream: nightly
      version: "4.11"
  latest:
    release:
      architecture: amd64
      channel: candidate
      version: "4.10"
  target:
    candidate:
      architecture: amd64
      product: ocp
      stream: nightly
      version: "4.12"
```
it means upgrade from 10->11->12. per @shellyyang1989 , pre will be done before last upgrade step, so it is 11.
so, I use 
```yaml
  source:
    candidate:
      architecture: amd64
      product: ocp
      stream: nightly
      version: "4.11"
```
not
```yaml
  source:
    candidate:
      architecture: amd64
      product: ocp
      stream: nightly
      version: "4.10"
```

Reply:
This job is cpou(control-plane-only update), which is not the same as chain upgrade(pre will be done before last upgrade step). So we should use the source version(`latest`) because preupgrade test is executed before cpou upgrade.

Reply:
> Now, there is change for pre-upgrade step: it is executed on final-1 version cluster.
> so, we need to inject oc according this version, or else some operators are not supported by old oc.

That is not exact, we can check `openshift-upgrade-qe-test-cpou` test chain, `openshift-extended-upgrade-pre` is executed on the N version, but not  final-1 version cluster. The case you mentioned only happened on chain upgrade nightly ci jobs.

Reply:
ok, got it. it is not same to what @shellyyang1989 told me.
I will update them per your comments.

Reply:
I am thinking if can introduce a new pre-upgrade step using a different version of cli injection, then reference it in chain upgrade yaml file, then we do not have to set the new release image everywhere.